apps/auth0-rules/src/add-user-metadata.ts

```diff
@@ -1,10 +1,63 @@
 import type { User } from '@asap-hub/auth';
-import type { UserMetadataResponse } from '@asap-hub/model';
+import type { gp2, UserMetadataResponse, UserResponse } from '@asap-hub/model';
 import got from 'got';
 import { URL, URLSearchParams } from 'url';
 import { handleError } from './handle-error';
 import type { Rule } from './types';
 
+type Auth0UserResponse = UserMetadataResponse | gp2.UserResponse;
+
+function isUserResponse(
+  response: UserMetadataResponse | gp2.UserResponse,
+): response is UserResponse & { algoliaApiKey: string } {
+  return 'onboarded' in (response as UserMetadataResponse);
+}
+
+const extractUser = (response: Auth0UserResponse) => {
```

Comment:
for helper methods like this i found that adding both input and output type make it easier to understand what they actually do, especially with a name as vague as this one
```suggestion
const extractUser = (response: Auth0UserResponse): User => {
```

Reply:
done